### PR TITLE
Prac 06 feedback code review request

### PR DESCRIPTION
**Pull request by DanL-Smith**
Hello @ChrisMarriner, please check formatting, naming and logic.

Had to close other branch due to commits being authored by my personal account. Sorry for any confusion

### CODE REVIEW

File: prac_06/guitars.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+def print_guitars(guitars):
+    """Print information about stored guitars"""
+    for i, guitar in enumerate(guitars, 1):
+        vintage_string = "(Vintage)" if guitar.is_vintage() else ""
+        print(
+            f"Guitar {i}: {guitar.name:{max(len(guitar.name) for guitar in guitars)}} ({guitar.year}), worth ${guitar.cost:.2f} {vintage_string}")
+
+
+def get_guitars():
```
Comment (lindsaymarkward) [inline prac_06/guitars.py:20]: Nice use of functions. `main` is easy to read.

File: prac_06/languages.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+"""CP1404/CP5632 Practical - language Program"""
+from prac_06.programming_language import ProgrammingLanguage
+
+
+ruby = ProgrammingLanguage("Ruby", "Dynamic", True, 1995)
+python = ProgrammingLanguage("Python", "Dynamic", True, 1991)
+visual_basic = ProgrammingLanguage("Visual Basic", "Static", False, 1991)
+programming_languages = [ruby, python, visual_basic]
+print("The dynamically typed languages are:")
+[print(language.name) if language.is_dynamic() else "" for language in programming_languages]
```
Comment (lindsaymarkward) [inline prac_06/languages.py:10]: This works, but is not really how to use a list comprehension. It creates a list containing things like `[None, "", None]`. It's harder to read. I'd suggest not doing this.

File: prac_06/programming_language.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    def __init__(self, name='', typing="", reflection="", year=0):
+        """Initialize a programming language instance"""
+        self.name = name
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+
+    def is_dynamic(self):
+        """Determine if instance is dynamic"""
+        return True if self.typing.lower() == "dynamic" else False
```
Comment (lindsaymarkward) [inline prac_06/programming_language.py:15]: See https://github.com/CP1404/Starter/wiki/Programming-Patterns#working-with-booleans

File: prac_06/guitars.py (hunk 1 above)
Comment (DanL-Smith) [reply prac_06/guitars.py:20]: Thank you for the feedback.

File: prac_06/languages.py (hunk 2 above)
Comment (DanL-Smith) [reply prac_06/languages.py:10]: Thanks for noting this. I have changed this line in to for loop.

File: prac_06/programming_language.py (hunk 3 above)
Comment (DanL-Smith) [reply prac_06/programming_language.py:15]: Ah. I noticed other students making this error too. Thanks for pointing this out.

Comment (lindsaymarkward) [reply prac_06/programming_language.py:15]: It's not an "error", but rather, there are several ways of doing it, and I think the easiest and most efficient is to return the result of the expression.

**Pull request merged**